### DISCUSSION TITLE
custom list function

### DISCUSSION THREAD

**Discussion by antonilol** [QuorumDMS/ftp-srv#260] [Q&A] [answered]
How do I tell the client that there are 'non existent' files with the functions `FileSystem.list`, `FileSystem.read` and `FileSystem.get`? For example a file called 'lastlogin' in the root of the server that contains the date of the last login of the user.

**Comment by matt-forster** [comment, marked as answer]
You'd have to return that file as a dynamic addition to the list/read/ get functions - probably handling that filename at the time of those calls specifically, and return the information for the file that way. 

That is pretty easy to do, but you'll have to make sure you return the correct types - using something like https://github.com/streamich/memfs helps with that.
